WordPress/gutenberg
# Global style revisions: show change summary on selected item

## What?

Maybe one day resolves https://github.com/WordPress/gutenberg/issues/55647

Part of:

- https://github.com/WordPress/gutenberg/issues/55776



https://github.com/WordPress/gutenberg/assets/6458278/0c57e6f5-0cf7-40f1-a750-ec40dbbbfd92




## Why?

To give the user a short overview of the global styles revision change set, which may not be evident from the canvas preview.

Also, highlighting revisions that match current editor styles by a left-hand-side dark, border.

## TODO


## How?

1. Deep comparison of two objects to get changed values
2. Limiting reported changes to the second level, e.g., `Block name > style`
3. Deduping results and translating
4. Caching results 

## Testing Instructions

Make a bunch of changes to your global styles revisions, ensuring to get a decent mix of block, top-level, element styles and layout, typography and color settings. 

If the theme, e.g., 2024, has style variations, activate and save a few revision using those.

Open up the styles revisions panel and check that the change summary makes sense.
